tck/methods/key_test.go

```diff
@@ -0,0 +1,182 @@
+package methods
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashgraph/hedera-sdk-go/tck/param"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerateKeyWithInvalidFromKey(t *testing.T) {
+	// Given
+	params := param.KeyParams{
+		Type:      param.ED25519_PRIVATE_KEY,
+		FromKey:   "someKey",
+		Threshold: 0,
+		Keys:      nil,
+	}
+
+	// When
+	_, err := GenerateKey(context.Background(), params)
+
+	// Then
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "invalid parameters: fromKey should only be provided for ed25519PublicKey, ecdsaSecp256k1PublicKey, or evmAddress types")
+}
+
+func TestGenerateKeyWithInvalidThreshold(t *testing.T) {
+	// Given
+	params := param.KeyParams{
+		Type:      param.ED25519_PUBLIC_KEY,
+		FromKey:   "",
+		Threshold: 1,
+		Keys:      nil,
+	}
+
+	// When
+	_, err := GenerateKey(context.Background(), params)
+
+	// Then
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "invalid parameters: threshold should only be provided for thresholdKey types")
```

Comment:
Maybe we should think of a way to centralise the error messages. As a changes for the error in one place for example the `GenerateKey` may lead to changes in the tests and overall test inconsistency. I personally don't like checking against concrete strings.

Reply:
Moved the errors to utils/key.go